pallets/foreign-investments/src/entities.rs

```diff
@@ -52,16 +56,31 @@ pub struct InvestmentInfo<T: Config> {
 	/// General info
 	pub base: BaseInfo<T>,
 
-	/// Total swapped amount pending to execute for decreasing the investment.
-	/// Measured in foreign currency
-	pub decrease_swapped_amount: T::Balance,
+	/// Pool invested amount, denominated in foreign currency.
+	///
+	/// Used to correlate the pool amount into foreign amount when the market
+	/// conversion is not known upfront.
+	pub invested_foreign_amount: T::Balance,
+
+	/// Total decrease pool amount pending to be swapped, denominated in foreign
+	/// currency.
+	///
+	/// Used to correlate the pool amount into foreign amount when the market
+	/// conversion is not known upfront.
+	pub decrease_pending_foreign_amount: T::Balance,
```

Comment:
README FIRST

I've added these two new variables to be able to correlate the foreign and pool amounts. This is the key idea of this refactor, and we need them to know the other amount upfront without knowing the market rate.

Basically, the idea is: I invest `100` of foreign, and later it becomes into `200` of the pool. I save those `100` as `invest_foreign_amount` to later correlate with the real `invested_pool_amount` and be able to know how much pool amount I need to decrease if later the user calls `decrease(50)`. For this example, it will be: 
```rust
pool_investment_decrement =
= (foreign_amount / invested_foreign_amount) * invested_pool_amount =
= (50 / 100) * 200 = 100
```

So, no matter how the market ratio changes, if later the user calls `decrease(50)`, the amount of pool that is no longer invested is `100`.